pkg/metrics/exporter.go

```diff
@@ -53,7 +92,7 @@ func NewMetricsExporter() error {
 	return nil
 }
 
-func getCurMetricsExporter() view.Exporter {
+func (mm *Manager) getCurMetricsExporter() view.Exporter {
```

Comment:
It doesn't look like we need to be swapping exporters. We can simplify this code considerably by avoiding this cache/retrieval mechanism and just running the exporter and returning.

Reply:
i think we should keep this as we want to implement more exporters over time like ocagent

Reply:
Leaving room for multiple exporters is fine. What seems unnecessary is the need to live-swap exporters, as they should have roughly the same lifespan as the parent process which means:

* No need for getCurMetricsExporter(), just create all metrics exporters on init
* No need for mutexes, startup is single-threaded
* No need for getCurPromSrv() + resetCurPromSrv() and the associated mutexes

Reply:
sounds good, removed getCurPromSrv, resetCurPromSrv, getCurMetricsExporter